src/cow-react/modules/limitOrders/containers/LimitOrdersConfirmModal/index.tsx

```diff
@@ -0,0 +1,60 @@
+import Modal from '@src/components/Modal'
+import React, { useCallback } from 'react'
+import { Currency, CurrencyAmount } from '@uniswap/sdk-core'
+import { CloseIcon } from '@src/theme'
+import { useAtomValue } from 'jotai/utils'
+import { useHigherUSDValue } from 'hooks/useStablecoinPrice'
+import { CurrencyInfo } from '@cow/common/pure/CurrencyInputPanel/typings'
+import { LimitOrdersConfirm } from '../../pure/LimitOrdersConfirm'
+import { tradeFlow, TradeFlowContext } from '../../services/tradeFlow'
+import { limitRateAtom } from '../../state/limitRateAtom'
+import * as styledEl from './styled'
+
+export interface LimitOrdersConfirmModalProps {
+  isOpen: boolean
+  tradeContext: TradeFlowContext
+  inputCurrencyInfo: CurrencyInfo
+  outputCurrencyInfo: CurrencyInfo
+  onDismiss(): void
+}
+
+function getCurrencyAmount(currency: Currency | null, value: string | null): CurrencyAmount<Currency> | null {
+  if (!currency) return null
+
+  // TODO: think about BigNumber usage
+  return CurrencyAmount.fromRawAmount(currency, Number(value || '0') * 10 ** currency.decimals)
```

Comment:
We definitely need to think about this as I'm not comfortable with this exponential operation using JS numbers 😰

Reply:
We will cooperate with @nenadV91 and refactor rate-related values to BigNumber